new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1056 @@
+        "Create configmap for domain creation failed");
+
+    // create a V1Container with specific scripts and properties for creating domain
+    V1Container jobCreationContainer = new V1Container()
+        .addCommandItem("/bin/sh")
+        .addArgsItem("/u01/oracle/oracle_common/common/bin/wlst.sh")
+        .addArgsItem("/u01/weblogic/" + wlstScriptFile.getFileName()) //wlst.sh
+        // script
+        .addArgsItem("-skipWLSModuleScanning")
+        .addArgsItem("-loadProperties")
+        .addArgsItem("/u01/weblogic/" + domainPropertiesFile.getFileName());
+    //domain property file
+
+    logger.info("Running a Kubernetes job to create the domain");
+    createDomainJob(pvName, pvcName, domainScriptConfigMapName, namespace, jobCreationContainer);
+
+  }
+
+  /**
+   * Create configmap containing domain creation scripts.
+   *
+   * @param configMapName name of the configmap to create
+   * @param files files to add in configmap
+   * @param namespace name of the namespace in which to create configmap
+   * @throws IOException when reading the domain script files fail
+   * @throws ApiException if create configmap fails
+   */
+  private void createConfigMapForDomainCreation(String configMapName, List<Path> files, String namespace)
+      throws ApiException, IOException {
+    logger.info("Creating configmap {0}", configMapName);
+
+    Path domainScriptsDir = Files.createDirectories(
+        Paths.get(TestConstants.LOGS_DIR, this.getClass().getSimpleName(), namespace));
+
+    // add domain creation scripts and properties files to the configmap
+    Map<String, String> data = new HashMap<>();
+    for (Path file : files) {
+      logger.info("Adding file {0} in configmap", file);
+      data.put(file.getFileName().toString(), Files.readString(file));
+      logger.info("Making a copy of file {0} to {1} for diagnostic purposes", file,
+          domainScriptsDir.resolve(file.getFileName()));
+      Files.copy(file, domainScriptsDir.resolve(file.getFileName()));
+    }
+    V1ObjectMeta meta = new V1ObjectMeta()
+        .name(configMapName)
+        .namespace(namespace);
+    V1ConfigMap configMap = new V1ConfigMap()
+        .data(data)
+        .metadata(meta);
+
+    boolean cmCreated = assertDoesNotThrow(() -> createConfigMap(configMap),
+        String.format("Failed to create configmap %s with files %s", configMapName, files));
+    assertTrue(cmCreated, String.format("Failed while creating ConfigMap %s", configMapName));
+  }
+
+  /**
+   * Create a job to create a domain in persistent volume.
+   *
+   * @param pvName name of the persistent volume to create domain in
+   * @param pvcName name of the persistent volume claim
+   * @param domainScriptCM configmap holding domain creation script files
+   * @param namespace name of the domain namespace in which the job is created
+   * @param jobContainer V1Container with job commands to create domain
+   */
+  private void createDomainJob(String pvName,
```

Comment:
Shouldn't this be in utils?

Reply:
yeah lot of the code can be moved to utils. For now I will keep this here.
Later on I will create a JIRA for refactoring all IT classes which has these methods.